rust_src/src/symbols.rs

```diff
@@ -404,5 +404,42 @@ pub fn symbol_value(symbol: LispSymbolRef) -> LispObject {
     }
     val
 }
+/// Non-nil if VARIABLE has a local binding in buffer BUFFER.
+/// BUFFER defaults to the current buffer.
+#[lisp_fn(min = "1")]
+pub fn local_variable_p(variable: LispObject, buffer: LispBufferOrCurrent) -> bool {
+    let buf: LispBufferRef = buffer.into();
+    let mut symbol: LispSymbolRef = variable.into();
+
+    while symbol.is_alias() {
+        symbol = symbol.get_indirect_variable();
```

Comment:
`get_indirect_variable` is already a loop.

`let symbol = variable.get_indirect_variable();` is all you need.